### PR TITLE
Introducing Fabric Video

### PR DESCRIPTION
**Pull request by regiskuckaertz**
There's a new fluid250 template in town. This one existed in DFP for a while, so we're just bringing it back inside our codebase so it plays nicely within our ad ecosystem.

fluid-video

cc @guardian/commercial-dev @rich-nguyen

### CODE REVIEW

**Comment by humaislam** [comment]
hiya - what do you need to get this shipped or closed @regiskuckaertz

**Comment by regiskuckaertz** [comment]
@humaislam I'm ready to go, just need approval from someone.

File: static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    function Fluid250VideoBg(adSlot, params) {
+        var isUpdating = false;
+        var hasVideo, video, layer2, inView;
+
+        adSlot = adSlot instanceof HTMLElement ? adSlot : adSlot[0];
+        fluid250VideoBgTpl || (fluid250VideoBgTpl = template(fluid250VideoBgStr));
+
+        hasVideo = !detect.isIOS();
+        if (hasVideo) {
+            params.videoOrPoster = '<video webkit-playsinline muted class="creative__video creative__video--' + params.Videoalignment + '"><source src="' + params.VideoURL + '" type="video/mp4"></video>';
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js]: could `videoOrPoster` be two separate params?

File: static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        } else {
+            params.videoOrPoster = '<div class="poster" style="background-image:url(' + params.VideoBackupImage + ')"></div>';
+        }
+
+        params.Layer2clickbuttonMarkup = params.Layer2clickbutton ?
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js]: Is it a bit odd to pass html markup as params to a template? Shouldn't the html be housed in the template?

File: static/src/stylesheets/module/commercial/creatives/_fabric-video.scss [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        clear: both;
+        position: absolute;
+    }
+
+    .mp4Video_centered {
+        left: 50%;
+        margin-left: -960px;
+    }
+
+    @media screen and (min-width: 740px) and (max-width: 1920px) {
```
Comment (rich-nguyen) [inline static/src/stylesheets/module/commercial/creatives/_fabric-video.scss]: no breakpoint mixin?

**Comment by rich-nguyen** [comment]
👍

File: static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js (hunk 1 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js]: I don't think so, it's really one or the other.

File: static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js (hunk 2 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js]: In that case, yes. Normally you can include logic in templates, but I have experienced problems with the `text!` plugin with expressions like `<%if(xxx === 'test'){%>`. Tests for the truthy value of an object value works though. Thanks!

File: static/src/stylesheets/module/commercial/creatives/_fabric-video.scss (hunk 3 above)
Comment (regiskuckaertz) [inline static/src/stylesheets/module/commercial/creatives/_fabric-video.scss]: oh yes, very good!

File: static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js (hunk 1 above)
Comment (jamesgorrie) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js]: Why not just call it `params.template` or `params.html`?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/fabric-video.js]: I feel it's a bit too abstract, but have renamed it `video` then.

**Pull request merged**